src/coreclr/md/enc/stgio.cpp

```diff
@@ -312,13 +312,14 @@ HRESULT StgIO::Open(                    // Return code.
     m_fFlags = fFlags;
     if ((szName != NULL) && (*szName != 0))
     {
-        WCHAR rcExt[_MAX_PATH];
-        SplitPath(szName, NULL, 0, NULL, 0, NULL, 0, rcExt, _MAX_PATH);
-        if (SString::_wcsicmp(rcExt, W(".obj")) == 0)
+        LPCWSTR ext;
+        size_t extSize;
+        SplitPathInterior(szName, NULL, 0, NULL, 0, NULL, 0, &ext, &extSize);
+        if (SString::_wcsicmp(ext, W(".obj")) == 0)
         {
             m_FileType = FILETYPE_NTOBJ;
```

Comment:
We can delete this special casing of .obj and .tlb files, and the matching `FILETYPE_NTOBJ` and `FILETYPE_TLB` enums. Nothing in .NET Core should be using these. (It can be a separate PR.)